src/image/transform/mask/yen.js

```diff
@@ -0,0 +1,50 @@
+// see https://github.com/fiji/Auto_Threshold/blob/master/src/main/java/fiji/threshold/Auto_Threshold.java
+// Implements Yen  thresholding method
+// 1) Yen J.C., Chang F.J., and Chang S. (1995) "A New Criterion
+//    for Automatic Multilevel Thresholding" IEEE Trans. on Image
+//    Processing, 4(3): 370-378
+// 2) Sezgin M. and Sankur B. (2004) "Survey over Image Thresholding
+//    Techniques and Quantitative Performance Evaluation" Journal of
+//    Electronic Imaging, 13(1): 146-165
+//    http://citeseer.ist.psu.edu/sezgin04survey.html
+//
+// M. Emre Celebi
+// 06.15.2007
+// Ported to ImageJ plugin by G.Landini from E Celebi's fourier_0.8 routines
+
+export default function yen(histogram, total) {
+    let threshold;
+    let crit;
+    let max_crit;
+    let norm_histo = []; /* normalized histogram */
+    let P1 = []; /* cumulative normalized histogram */
+    let P1_sq = [];
+    let P2_sq = [];
+
+    for (let ih = 0; ih < histogram.length; ih++)
+        norm_histo[ih] = histogram[ih] / total;
+
+    P1[0] = norm_histo[0];
+    for (let ih = 1; ih < histogram.length; ih++)
+        P1[ih] = P1[ih - 1] + norm_histo[ih];
+
+    P1_sq[0] = norm_histo[0] * norm_histo[0];
+    for (let ih = 1; ih < histogram.length; ih++)
+        P1_sq[ih] = P1_sq[ih - 1] + norm_histo[ih] * norm_histo[ih];
+
+    P2_sq[histogram.length - 1] = 0.0;
+    for (let ih = histogram.length - 2; ih >= 0; ih--)
+        P2_sq[ih] = P2_sq[ih + 1] + norm_histo[ih + 1] * norm_histo[ih + 1];
+
+    /* Find the threshold that maximizes the criterion */
+    threshold = -1;
```

Comment:
I would prefer if you declare the variable when you first need it (if possible).
`let threshold = -1;` here and remove the declaration at the beginning of the function. It is clearer that way because we directly know that you don't use the variable before.